docs/spec/distributing.rst

```diff
@@ -266,3 +358,60 @@ of that Python version. This can be queried e.g.
 ``pythonX.Y -c 'import site; print(site.getsitepackages())'``. It is also recommended
 that the type checker allow for the user to point to a particular Python
 binary, in case it is not in the path.
+
+.. _library-interface:
+
+Library interface (public and private symbols)
+----------------------------------------------
+
+If a ``py.typed`` module is present, a type checker will treat all modules
+within that package (i.e. all files that end in ``.py`` or ``.pyi``) as
+importable unless the file name begins with an underscore. These modules
+comprise the supported interface for the library.
+
+Each module exposes a set of symbols. Some of these symbols are
+considered "private” — implementation details that are not part of the
+library’s interface. Type checkers can use the following rules
+to determine which symbols are visible outside of the package.
+
+-  Symbols whose names begin with an underscore (but are not dunder
```

Comment:
What about objects `@typing.type_check_only`? Are they exposed?

Reply:
`typing.type_check_only` is orthogonal to whether a symbol should be considered public or private. It means that a type checker should report an error if that symbol is used in a runtime context. These symbols can still be used in type expressions if they are publicly exposed from a library or stub file.